### PR TITLE
LLD - Stax OS update Pixel Polish 

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Small modifications made to stax os update flow on Desktop

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10648] [LIVE-10651] [LIVE-10669] [LIVE-10672]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10648]: https://ledgerhq.atlassian.net/browse/LIVE-10648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10651]: https://ledgerhq.atlassian.net/browse/LIVE-10651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10669]: https://ledgerhq.atlassian.net/browse/LIVE-10669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10672]: https://ledgerhq.atlassian.net/browse/LIVE-10672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx [hunk 1]
```diff
@@ -106,17 +107,31 @@ const UpdateModal = ({
   const withFinal = useMemo(() => hasFinalFirmware(firmware?.final), [firmware]);
   const [cancel, setCancel] = useState<boolean>(false);
 
+  const isDisconnectedDeviceError = err instanceof DisconnectedDevice;
+  const isDisconnectedDeviceDuringOperationError = err instanceof DisconnectedDeviceDuringOperation;
+  const isDeviceDisconnected =
+    isDisconnectedDeviceError || isDisconnectedDeviceDuringOperationError;
+
+  console.log("DEVICE", deviceModelId, props.device, updatedDeviceInfo);
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:115]: 🗑️

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx [hunk 2]
```diff
@@ -106,17 +107,31 @@ const UpdateModal = ({
...
-      const restoreStepLabel = t("manager.modal.steps.restore");
-      const installUpdateLabel =
+
+      const restoreStepLabel =
         stateStepId === "finish"
+          ? t("manager.modal.steps.restoreDone")
+          : t("manager.modal.steps.restore");
+
+      const installUpdateLabel =
+        stateStepId === "finish" || stateStepId === "restore"
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:134]: I think that we could improve how we deal with constant values with TS. Could be something like:
```typescript
const STEPS = {
  RESET_DEVICE: "resetDevice",
  ID_CHECK: "idCheck",
  UPDATE_MCU: "updateMCU",
  UPDATING: "updating",
  RESTORE: "restore",
  FINISH: "finish",
} as const;

export type StepId = (typeof STEPS)[keyof typeof STEPS];

...
const installUpdateLabel = stateStepId === STEPS.FINISH ...

```

wdyt?

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx [hunk 3]
```diff
@@ -106,17 +107,31 @@ const UpdateModal = ({
...
+  }, [cancel, showDisclaimer, isDeviceDisconnected, stateStepId, onRequestClose]);
 
   const createSteps = useCallback(
     ({ withResetStep }: { withResetStep: boolean }) => {
       const hasRestoreStep =
         firmware && isDeviceLocalizationSupported(firmware.final.name, deviceModelId);
-      const restoreStepLabel = t("manager.modal.steps.restore");
-      const installUpdateLabel =
+
+      const restoreStepLabel =
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:128]: I know it was already done like this, but I think we should extract this logic outside of this component with a function that builds steps.

Comment (KVNLS) [reply apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:128]: We don't need a translation function at this stage, we could do it later in the component rendering:
label={t(step.label) as string}

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx (hunk 2 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:134]: Yes that would be much cleaner, I'll make the modifications

**Review by KVNLS — APPROVED** [review]
Great job!

**Pull request merged**